examples/interaction.rs

```diff
@@ -0,0 +1,292 @@
+        .actor(Choice::new(ExternalInput {
+            threshold: 3,
+            supervisor_addr: 1.into(),
+        }))
+        .actor(
+            Choice::new(Supervisor {
+                initial_state: SupervisorState {
+                    addr: 1.into(),
+                    threshold: 3,
+                    counter_addr: 2.into(),
+                    input_addr: 0.into(),
+                    success: false,
+                },
+            })
+            .or(),
+        )
+        .actor(
+            Choice::new(Counter {
+                initial_state: CounterState {
+                    addr: 1.into(),
+                    counter: 0,
+                },
+            })
+            .or()
+            .or(),
+        );
+
+    match args.subcommand()?.as_deref() {
+        Some("check") => {
+            let checker = model
+                .property(Expectation::Eventually, "success", |_, state| {
+                    state.actor_states.iter().any(|s| state_filter_success(s))
+                })
+                .checker()
+                .threads(num_cpus::get())
+                .target_max_depth(30) // To ensure termination of the checker, since the state space is very loosely bounded.
+                .spawn_bfs()
+                .report(&mut WriteReporter::new(&mut std::io::stdout()));
+
+            checker.assert_properties();
+        }
+
+        Some("explore") => {
+            let checker = model
+                .property(Expectation::Eventually, "success", |_, state| {
+                    state.actor_states.iter().any(|s| state_filter_success(s))
+                })
+                .checker()
+                .threads(num_cpus::get())
+                .target_max_depth(30); // To ensure termination of the checker, since the state space is very loosely bounded.
+
+            checker.serve("0:3000");
+        }
+
+        _ => {
+            println!("USAGE:");
+            println!("  ./interaction check");
+            println!("  ./interaction explore");
+        }
+    }
+    Ok(())
+}
+
+// We use this helper function to improve readability of check_discovery.
+// It also shows how to use the choice! macro for destructuring.
+fn state_filter_success(s: &Arc<choice![InputState, SupervisorState, CounterState]>) -> bool {
+    match s.as_ref() {
+        choice!(0 -> _v) => false,    // InputState
+        choice!(1 -> v) => v.success, // SupervisorState
+        choice!(2 -> _v) => false,    // CounterState
+        choice!(3 -> !) => false,     // Never
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub enum Msg {
+    SupervisorIncrementRequest(CounterSize),
+    SupervisorReportRequest(),
+    CounterReplyCount(CounterSize),
+}
+
+// We use a Generic BaseActor to pass State to on_start, but Actors could be built using independent structs.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct BaseActor<T> {
+    pub initial_state: T, //T needs to be a State Type
+}
```

Comment:
I recommend removing this in favor of distinct structs for simplicity.

Reply:
If we remove the supervisor, as mentioned further down, there is no need for this anyways, so yes!